base/strings/basic.jl

```diff
@@ -34,6 +34,7 @@ If you need to subsequently modify `v`, use `String(copy(v))` instead.
 """
 String(v::Array{UInt8,1})
 
+String(x::Symbol) = string(x)
```

Comment:
Can we make this `convert(::Type{String}, x::Symbol)` for consistency? We already have `convert(Symbol, ::String)`. It would be weird if `Symbol(str)` and `s::Symbol = str` works, and `String(sym)` works, but `x::String = sym` does not.